wgpu-types/src/lib.rs

```diff
@@ -1443,6 +1465,116 @@ impl<L> SamplerDescriptor<L> {
     }
 }
 
+/// Resource that can be bound to a pipeline.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BindingResource<'a, Buffer, Sampler, TextureView, TextureViewArrayElement> {
```

Comment:
ok, I think this is going a bit too far :)
Let's just have separate types here, we don't want to use more than a few generics